src/core/BufferGeometry.js

```diff
@@ -642,9 +642,49 @@ Object.assign( BufferGeometry.prototype, EventDispatcher.prototype, {
 
 	}(),
 
+	/**
+	 * Allows flat shading of faces on non-indexed buffer geometry
+	 */
 	computeFaceNormals: function () {
 
-		// backwards compatibility
+		var pos = this.attributes.position.array;
+		var attrName = 'normal';
+		if ( this.attributes[ attrName ] === undefined ) {
+
+			this.addAttribute( attrName, new THREE.BufferAttribute( new Float32Array( pos.length ), 3 ) );
+
+		}
+		var n = this.attributes[ attrName ].array;
+		var i, v, il;
+		var cb = new THREE.Vector3();
+		var ab = new THREE.Vector3();
+		for ( i = 0, il = pos.length; i < il; i += 9 ) {
+
+			var iA = i;
+			var iB = i + 3;
+			var iC = i + 6;
+
+			cb.set( pos[ iC ] - pos[ iB ],
+			pos[ iC + 1 ] - pos[ iB + 1 ],
+			pos[ iC + 2 ] - pos[ iB + 2 ] );
+			ab.set( pos[ iA ] - pos[ iB ],
+			pos[ iA + 1 ] - pos[ iB + 1 ],
+			pos[ iA + 2 ] - pos[ iB + 2 ] );
+			cb.cross( ab );
+			cb.normalize();
+
+			var face = [ iA, iB, iC ];
+			// For each vertex on the face
```

Comment:
I would not instantiate an array in this case just for the purpose of being able to loop over it. Just assign the 9 values.

In any event, I am not sure this method is needed.